### ISSUE TITLE
[sct_segment_GM] temp folder deletion error

### ISSUE THREAD

**Issue by tanguyduval** [neuropoly/spinalcordtoolbox#1263]
Bug can be reproduced on my CentOs station

````
  Split along rostro-caudal direction...
Traceback (most recent call last):
  File "/autofs/cluster/msilab/users/tools/Spine/SpinalCordToolbox/scripts/sct_segment_graymatter.py", line 760, in <module>
    main()
  File "/autofs/cluster/msilab/users/tools/Spine/SpinalCordToolbox/scripts/sct_segment_graymatter.py", line 754, in main
    seg_gm.segment()
  File "/autofs/cluster/msilab/users/tools/Spine/SpinalCordToolbox/scripts/sct_segment_graymatter.py", line 232, in segment
    self.target_im, self.info_preprocessing = pre_processing(self.param_seg.fname_im, self.param_seg.fname_seg, self.param_seg.fname_level, new_res=self.param_data.axial_res, square_size_size_mm=self.param_data.square_size_size_mm, denoising=self.param_data.denoising, verbose=self.param.verbose, rm_tmp=self.param.rm_tmp)
  File "/autofs/cluster/msilab/users/tools/Spine/SpinalCordToolbox/scripts/msct_gmseg_utils.py", line 191, in pre_processing
    shutil.rmtree(tmp_dir)
  File "/autofs/cluster/msilab/users/tools/Spine/SpinalCordToolbox/python/lib/python2.7/shutil.py", line 256, in rmtree
    onerror(os.rmdir, path, sys.exc_info())
  File "/autofs/cluster/msilab/users/tools/Spine/SpinalCordToolbox/python/lib/python2.7/shutil.py", line 254, in rmtree
    os.rmdir(path)
OSError: [Errno 39] Directory not empty: 'tmp_preprocessing_170418200540_636677/'
````

**Comment by tanguyduval** [comment]
Note that using -r 0 works great. Perfect segmention BTW.

**Comment by jcohenadad** [comment]
fixed in PR #1535
